docs/workflow/building/coreclr/android.md

```diff
@@ -1,100 +1,172 @@
-`adb`, and then run them in an interactive Android shell using `adb shell`:
+To build `HelloAndroid`, run the following command from `<repo_root>`:
 
-To copy the PAL tests over to an Android phone:
 ```
-adb push artifacts/obj/coreclr/Linux.arm64.Debug/src/pal/tests/palsuite/ /data/local/tmp/coreclr/pal/tests/palsuite
-adb push cross/android/toolchain/arm64/sysroot/usr/lib/libandroid-support.so /data/local/tmp/coreclr/lib/
-adb push cross/android/toolchain/arm64/sysroot/usr/lib/libandroid-glob.so /data/local/tmp/coreclr/lib/
-adb push src/pal/tests/palsuite/paltestlist.txt /data/local/tmp/coreclr
-adb push src/pal/tests/palsuite/runpaltests.sh /data/local/tmp/coreclr/
+make BUILD_CONFIG=<Debug|Release> TARGET_ARCH=<x64|arm64> RUNTIME_FLAVOR=CoreCLR DEPLOY_AND_RUN=false run -C src/mono/sample/Android
 ```
 
-Then, use `adb shell` to launch a shell on Android. Inside that shell, you can launch the PAL tests:
+On successful execution, the command will output the `HelloAndroid.apk` at:
 ```
-LD_LIBRARY_PATH=/data/local/tmp/coreclr/lib ./runpaltests.sh /data/local/tmp/coreclr/
+<repo-root>artifacts/bin/AndroidSampleApp/<x64|arm64>/<Debug|Release>/android-<x64|arm64>/Bundle/bin/HelloAndroid.apk
 ```
 
-Debugging coreclr on Android
-----------------------------
+### Running HelloAndroid sample on an emulator
 
-You can debug coreclr on Android using a remote lldb server which you run on your Android device.
+To run the sample on an emulator, the emulator first needs to be up and running.
 
-First, push the lldb server to Android:
+Creating an emulator (ADV - Android Virtual Device) can be achieved through [Android Studio - Device Manager](https://developer.android.com/studio/run/managing-avds).
 
+After its creation, the emulator needs to be booted up and running, so that we can run the `HelloAndroid` sample on it via:
 ```
-adb push cross/android/lldb/2.2/android/arm64-v8a/lldb-server /data/local/tmp/
+make BUILD_CONFIG=<Debug|Release> TARGET_ARCH=<x64|arm64> RUNTIME_FLAVOR=CoreCLR DEPLOY_AND_RUN=true run -C src/mono/sample/Android
 ```
 
-Then, launch the lldb server on the Android device. Open a shell using `adb shell` and run:
-
+NOTE: Emulators can be also started from the terminal via:
 ```
-adb shell
-cd /data/local/tmp
-./lldb-server platform --listen *:1234
+$ANDROID_SDK_ROOT/emulator/emulator -avd <emulator-name>
 ```
 
-After that, you'll need to forward port 1234 from your Android device to your PC:
+#### WSL2
+
+The app can be run on an emulator running on the Windows host.
+1. Install Android Studio on the Windows host (same versions as in [prerequisites](#prerequisite))
+2. In Windows, create and start an emulator
+3. In WSL, swap the `adb` from the Android SDK in WSL2 with that from Windows
+    - `mv $ANDROID_SDK_ROOT/platform-tools/adb $ANDROID_SDK_ROOT/platform-tools/adb-orig`
+    - `ln -s /mnt/<path-to-adb-on-host> $ANDROID_SDK_ROOT/platform-tools/adb`
```

Comment:
do you really need to symlink or would it be enough to set xharness ADB_EXE_PATH to `/mnt/<path-to-adb-on-host>` ?

Reply:
@elinor-fung ^

Reply:
I had to symlink it somewhere. It didn't necessarily need to be the path from the Android SDK in WSL2, but I found that convenient for just using `adb` from the command line in WSL.

When I directly set `ADB_EXE_PATH=/mnt/<path-to-adb-on-host>`, running with xharness would fail with:
```
Exit code: 1
Std out:
Performing Streamed Install
        
Std err:
adb: failed to stat /home/elinor/runtime/artifacts/bin/AndroidSampleApp/x64/Release/android-x64/Bundle//bin/HelloAndroid.apk: No such file or directory
```

I admittedly didn't really spend time looking into why.

Directly running `/mnt/<path-to-adb-on-host> install <path-to-apk-in-wsl>` from my WSL terminal works fine, so I guess it may be something with how it gets launched with xharness?